### ISSUE TITLE
undefined fields shouldn't be used when checking enum

### ISSUE THREAD

**Issue by wavern1** [mafintosh/is-my-json-valid#142]
If a field is `undefined` and the schema specifies it as being an enum, it errors out saying `"must be an enum value"`.

I think the logical approach here would be to skip that field unless `isRequired` is set to `true` on it as well

**Comment by LinusU** [comment]
Yeah, the same is true when `type: ['null', 'string'], enum: ['a', 'b']`, `null` is not accepted. Unfortunately I think that this is a part of the JSON Schema specification so I'm not sure that we could change it.

If someone could quote the spec on how this should be handled I'd be happy to fix it 👍

**Comment by RangerMauve** [comment]
I think the main issue is that `undefined` isn't a valid JSON value, so the spec wouldn't have anything to say on the matter.

In my opinion, `undefined` should be treated the same as the key not existing, since it would disappear when the object got serialized to JSON.

**Comment by LinusU** [comment]
Ah, of course, that is true. Yeah, I also think that we should treat `undefined` as the value isn't there...

**Comment by RangerMauve** [comment]
It'll probably make calculating `minProperties` and `maxProperties` more annoying. :P
